t/90h2olog.t

```diff
@@ -0,0 +1,107 @@
+#!perl
+# DTRACE_TESTS=1 to skip to check prereqisites
+# H2OLOG_DEBUG=1 for more runtime logs
+use strict;
+use warnings FATAL => "all";
+use Net::EmptyPort qw(empty_port);
+use Test::More;
+use JSON;
+use t::Util;
+
+run_as_root();
+
+my $h2olog_prog = bindir() . "/h2olog";
+my $client_prog = bindir() . "/h2o-httpclient";
+
+unless ($ENV{DTRACE_TESTS})  {
+  plan skip_all => "$h2olog_prog not found"
+      unless -e $h2olog_prog;
+
+  plan skip_all => "$client_prog not found"
+      unless -e $client_prog;
+
+  plan skip_all => 'dtrace support is off'
+      unless server_features()->{dtrace};
+}
+
+my $quic_port = empty_port({
+    host  => "127.0.0.1",
+    proto => "udp",
+});
+
+my $server = spawn_h2o({
+    opts => [qw(--mode=worker)],
+    conf => << "EOT",
+listen:
+  type: quic
+  port: $quic_port
+  ssl:
+    key-file: examples/h2o/server.key
+    certificate-file: examples/h2o/server.crt
+hosts:
+  default:
+    paths:
+      /:
+        file.dir: t/assets/doc_root
+EOT
+});
+
+subtest "h2olog", sub {
+  my $tracer = spawn_h2olog({
+    pid => $server->{pid},
+    args => [$ENV{H2OLOG_DEBUG} ? ("-d") : ()],
+  });
+
+  my $retry = 10;
+  my $trace;
+  do {
+      sleep 1;
+      diag "making requests and tracing (#${retry})";
+
+      my ($headers, $body) = run_prog("$client_prog -3 https://127.0.0.1:$quic_port/");
+      like $headers, qr{^HTTP/3 200\n}, "req: HTTP/3";
+
+      if (--$retry < 0) {
```

Comment:
I'm not sure if I like this `retry` thing. While this approach might work for testing that `h2olog` emits something, it cannot be used to test anything more than that, when we aren't sure for which interaction h2olog is going to emit something.

Could we do something like following:
* let h2olog emit something (maybe a fake event or a line) that indicates that the probes are being attached, and
* spin within `spawn_h2olog` until that thing is being emitted?

Reply:
Agreed that this retry does little things in this case. However, I rather think of other cases when this loop waits for a particular event. So I'd remove this retry. What do you think?

Reply:
>*  let h2olog emit something (maybe a fake event or a line) that indicates that the probes are being attached

I've chosen it. Done in f5d372b